core/server/master/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -210,24 +211,26 @@
    * (A) cannot call (D)
    *
    * (B) private (or package private) methods that journal:
-   * These methods perform the work from the public apis, and also asynchronously write to the
-   * journal (for write operations). The names of these methods are suffixed with "AndJournal".
+   * These methods perform the work from the public apis, like checking and error handling, and also
+   * asynchronously append entries to the journal. The names of these methods are suffixed with
+   * "AndJournal".
    * (B) cannot call (A)
    * (B) can call (B)
    * (B) can call (C)
    * (B) cannot call (D)
    *
    * (C) private (or package private) internal methods:
-   * These methods perform the rest of the work, and do not do any journaling. The names of these
-   * methods are suffixed by "Internal".
+   * These methods perform the rest of the work, and may append to the journal. The names of these
```

Comment:
This is related to the fact that we could be journaling while holding the READ lock. Before, holding the WRITE lock allowed us to treat it as a big lock to protect multiple operations, so we could journal at any time while holding the WRITE lock.

Now that we update state while holding the READ lock, journaling operations should be appended when they happen, because other threads could view the mutated state. Therefore, there needs to be more fine-grained journal appending, so internal methods may need to journal along the way.

We still disallow `fromEntry` methods from journaling, by using the `NoopJournalContext` in those methods.